src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/NodeLabelMappingToLiteralsImpl.java

```diff
@@ -0,0 +1,25 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.impl;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.vocabulary.XSD;
+
+public class NodeLabelMappingToLiteralsImpl implements NodeLabelMapping{
+
+    @Override
+    public Node map(final String label) {
+        return NodeFactory.createLiteral(label);
+    }
+
+    @Override
+    public String unmap(final Node node) {
+        if (!node.isLiteral() || !node.getLiteral().getDatatypeURI().equals(XSD.xstring.getURI()))
```

Comment:
I realize only now that this is the same check as the one that you do in the next method. Therefore, it would be better (to avoid code duplication) to simply call the next method here instead of repeating the code of that method.

Can you please implement this change not only here but also in the other three classes (NodeLabelMappingToURIsImpl, NodeMappingToURIsImpl, and NodeMappingToBNodesImpl).